pages/studentPortal/labs/multiplication/game.tsx

```diff
@@ -115,32 +265,102 @@ export default function BlockComponentGallery() {
-          <Button label={"Next Player"} onClick={() => handlePlayer()} />
-          <Button label={"Reset Game"} onClick={() => handleReset()} />
-          <Button label={"Show Winner"} onClick={handleCalculateWinner} />
-          <h1 className="flex justify-center">Multiplication Game</h1>
-          {gameState.map((item, index) => (
-            <MultiplicationBlock
-              text={item.text}
-              onClick={() => handleSelect(index)}
-              blockState={item.state}
-            />
-          ))}
+        <div className="grid grid-cols-6 grid-rows-7">
+          <div className="pb-4 font-black col-start-1 col-end-6 flex justify-evenly w-[45rem]">
+            Current Player: {isPlayerOneActive ? "Player 1" : "Player 2"}
+          </div>
+          <div className="pb-8 col-start-1 col-end-7 flex justify-evenly w-[45rem]">
+            <Button label={"Next Player"} onClick={() => handlePlayer()} />
+            <Button label={"Reset Game"} onClick={() => handleReset()} />
+            <Button label={"Show Winner"} onClick={handleCalculateWinner} />
+          </div>
+          <div className="flex flex-row">
+            {gameState.slice(0, 9).map((item, index) => (
+              <MultiplicationBlock
+                text={item.text}
+                onClick={() => handleSelect(index)}
+                blockState={item.state}
+              />
+            ))}
+          </div>
+          <div className="grid grid-cols-9 col-span-6 w-[45rem]">
+            <div className="flex flex-col">
+              {gameState.slice(9, 20).map((item, index) => (
+                <MultiplicationBlock
+                  text={item.text}
+                  onClick={() => handleSelect(index)}
+                  blockState={item.state}
+                />
+              ))}
+            </div>
+            <div className="col-span-7 bg-blue-800">Image</div>
+            <div className="flex flex-col">
+              {gameState.slice(20, 31).map((item, index) => (
+                <MultiplicationBlock
+                  text={item.text}
+                  onClick={() => handleSelect(index)}
+                  blockState={item.state}
+                />
+              ))}
+            </div>
+          </div>
+          <div className="flex flex-row">
+            {gameState.slice(31, 41).map((item, index) => (
+              <MultiplicationBlock
+                text={item.text}
+                onClick={() => handleSelect(index)}
+                blockState={item.state}
+              />
+            ))}
+          </div>
         </div>
       ) : stage === STAGE.CALCULATE_WINNER ? (
-        <div className="flex justify-center">
-          <div className="flex flex-row justify-between "></div>
-          <Button
-            label={"Show Winner"}
-            onClick={() => calculateWinner(gameState)}
-          />
+        <div className="flex-row justify-center">
```

Comment:
can you extract the game over UI into its own component? the same with the rules component? 

instead of putting them all in the math folder, can you make a new folder inside of the math folder called `longestStreak` or if you have a better name for this game that is more descriptive than just `game`

Reply:
calling them SetRules and CalculateWinner and put them in the longestStreak folder